src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryService.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.cft.query;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.access.entities.AccessControlResponse;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.permission.entities.PermissionTypes;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.entities.TaskResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.repository.TaskResourceRepository;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.SearchTaskRequest;
+
+import java.util.List;
+
+@Service
+public class CftQueryService {
+
+    private final TaskResourceRepository taskResourceRepository;
+
+    public CftQueryService(TaskResourceRepository taskResourceRepository) {
+        this.taskResourceRepository = taskResourceRepository;
+    }
+
+    public List<TaskResource> getAllTasks(
+        SearchTaskRequest searchTaskRequest,
+        AccessControlResponse accessControlResponse,
+         List<PermissionTypes> permissionsRequired
+    ) {
+        final Specification<TaskResource> taskResourceSpecification = TaskResourceSpecification
+            .getTasks(searchTaskRequest, accessControlResponse, permissionsRequired);
+
+        Pageable page = PageRequest.of(0, 10, Sort.Direction.ASC, "taskId");
```

Comment:
So should this not take the page and size params from the incoming request also?

Reply:
Yes, it was hardcoded to test the pagination. But will change with RWA-591 ticket.

Reply:
Just leave it out as a rule if it's in another ticket.